### PR TITLE
Add a command line argument to simplify development of web extensions

### PR DESCRIPTION
**Pull request by aeschli**
Adds a new command line parameter `--extensionDevelopmentKind` to specify the extension kind of the extension(s) located at `--extensionDevelopmentPath`

The presence of  `--extensionDevelopmentKind` will start a web worker extension host (no setting needed) and will set the extension kind of the extension under dev to `web`

The new option can then be used in a launch config used for running or testing a web extension under development:

```
{
	"version": "0.2.0",
	"configurations": [
		{
			"name": "Run Web Extension",
			"type": "pwa-extensionHost",
			"request": "launch",
			"args": [
				"--extensionDevelopmentPath=${workspaceFolder}",
				"--extensionDevelopmentKind=web",
			],
```

Alternative approaches;
- command line arg `-webExtensionDevelopmentPath` to list the location(s) of web extensions

Note: The change also requires server code adoption

### CODE REVIEW

File: src/vs/workbench/api/common/extHost.protocol.ts [hunk 1]
```diff
@@ -69,6 +69,7 @@ export interface IEnvironment {
 	appLanguage: string;
 	appUriScheme: string;
 	extensionDevelopmentLocationURI?: URI[];
+	extensionDevelopmentKind?: ExtensionKind[];
```
Comment (alexdima) [inline src/vs/workbench/api/common/extHost.protocol.ts:72]: why does the kind need to travel to the extension host?

**Review by alexdima — CHANGES_REQUESTED** [review]
I am not a big fan of patching of the `package.json` in the scanning code. When we discussed, I understood that you will cover this via changing the `getExtensionKind` implementation (the other PR). The advantage of changing `getExtensionKind` is that it already abstracts the various ways in which the extension kind can be tweaked:
- from the extension's package.json
- from the product.json
- from the settings.json

I would have assumed that the command line option would be parsed and a map of extension id -> extension kind would be kept in `IEnvironmentService`. Then, this map could be retrieved by `getExtensionKind`.

The problem with patching the `package.json` is that the extensions viewlet will now disagree compared to the extension runtime because the extensions viewlet does its own extension scanning in the shared process and that one is not being patched (it would also be difficult because the shared process is shared across all windows, even those without extension development).

So overall, I think I would have expected to see a change in [`getExtensionKind`](https://github.com/microsoft/vscode/blob/ceb3bb2ce7653b032a3c6d86c11137f41c9e9947/src/vs/workbench/services/extensions/common/extensionsUtil.ts#L43), which is already adopted by both the extension viewlet (management) and the extension runtime.

**Comment by aeschli** [comment]
Thanks @alexdima, I'll look into a PR that does it using `getExtensionKind`

**Comment by aeschli** [comment]
Closing in favor of #118766

**Pull request closed without merging**